eng/pipelines/templates/steps/prepare-pipelines.yml

```diff
@@ -0,0 +1,30 @@
+parameters:
```

Comment:
Might need DevOpsPath as well if we want this to work for the private repos.

Reply:
I deliberately hadn't tackled private repos yet (I saw that as a refinement after I got this basic mechanism working in our public repos). To do private repos correctly I'd need to parse the repo name because I'd need to construct the DevOpsPath and Prefix from it (not difficult but I was aiming to get the basics right first!)

Reply:
Hey Wes, decided to do a cheap and cheerful approach to covering private repos using a YAML ```${{ if ... }}``` statement. One thing that I'm not sure about is whether we should bother creating the ```up``` and ```tests``` convention pipelines. The problem with both of these is that they'll get scheduled to run on a nightly basis.

This mechanism analyzes the entire repository so we'll get ```ci``` convention pipelines for all service directories the first time someone does this prepare step. I'm almost tempted to avoid supporting ```-pr``` repos entirely given the potential to create an overwhelming number of redundant pipelines.

Reply:
Perhaps we can disable the schedule triggers for anything not in the default devops path? Sort of like we don't have scheduled builds for anything not in master by default.

Reply:
I think I'd probably just add an extra option ```--no-schedule``` to Pipeline Generator so its explicit. What do you think?

Reply:
PR with ```--no-schedule``` option in Pipeline Generator: https://github.com/Azure/azure-sdk-tools/pull/1110

Reply:
Updated this PR with the --no-schedule option, so that once the other PR is approved and merged this can go in.

Reply:
I'm good with this approach at least for now.